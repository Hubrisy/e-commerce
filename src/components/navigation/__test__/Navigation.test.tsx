import React from 'react';
import { render } from '@testing-library/react';

import Navigation from '..';

jest.mock('@assets/svg/icons/phone.svg', () => {
  const PhoneIcon = () => <svg />;
  PhoneIcon.displayName = 'PhoneIcon';

  return PhoneIcon;
});
jest.mock('@assets/svg/icons/camera.svg', () => {
  const CameraIcon = () => <svg />;
  CameraIcon.displayName = 'CameraIcon';

  return CameraIcon;
});
jest.mock('@assets/svg/icons/headphones.svg', () => {
  const HeadphonesIcon = () => <svg />;
  HeadphonesIcon.displayName = 'HeadphonesIcon';

  return HeadphonesIcon;
});
jest.mock('@assets/svg/icons/computers.svg', () => {
  const ComputersIcon = () => <svg />;
  ComputersIcon.displayName = 'ComputersIcon';

  return ComputersIcon;
});
jest.mock('@assets/svg/icons/gaming.svg', () => {
  const GamingIcon = () => <svg />;
  GamingIcon.displayName = 'GamingIcon';

  return GamingIcon;
});

describe('Navigation component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Navigation />);
    expect(asFragment()).toMatchSnapshot();
  });
});
