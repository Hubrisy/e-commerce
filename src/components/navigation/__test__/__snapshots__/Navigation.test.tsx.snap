// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="w-full mt-[88px] bg-[#2E2E2E] h-[48px] text-granite-grey font-medium hidden items-center justify-center md:flex"
  >
    <div
      class="flex justify-center items-center"
    >
      <div
        class="flex items-center justify-center cursor-pointer"
      >
        <div>
          <svg />
        </div>
        <div
          class="ml-[8px]"
        >
          Phones
        </div>
        <div
          class="h-[24px] w-[1px] bg-granite-grey md:mx-[10px] lg:mx-[36px] xl:mx-4xlarge"
        />
      </div>
      <div
        class="flex items-center justify-center cursor-pointer"
      >
        <div>
          <svg />
        </div>
        <div
          class="ml-[8px]"
        >
          Computers
        </div>
        <div
          class="h-[24px] w-[1px] bg-granite-grey md:mx-[10px] lg:mx-[36px] xl:mx-4xlarge"
        />
      </div>
      <div
        class="flex items-center justify-center cursor-pointer"
      >
        <div>
          <svg />
        </div>
        <div
          class="ml-[8px]"
        >
          Smart watches
        </div>
        <div
          class="h-[24px] w-[1px] bg-granite-grey md:mx-[10px] lg:mx-[36px] xl:mx-4xlarge"
        />
      </div>
      <div
        class="flex items-center justify-center cursor-pointer"
      >
        <div>
          <svg />
        </div>
        <div
          class="ml-[8px]"
        >
          Cameras
        </div>
        <div
          class="h-[24px] w-[1px] bg-granite-grey md:mx-[10px] lg:mx-[36px] xl:mx-4xlarge"
        />
      </div>
      <div
        class="flex items-center justify-center cursor-pointer"
      >
        <div>
          <svg />
        </div>
        <div
          class="ml-[8px]"
        >
          Headphones
        </div>
        <div
          class="h-[24px] w-[1px] bg-granite-grey md:mx-[10px] lg:mx-[36px] xl:mx-4xlarge"
        />
      </div>
      <div
        class="flex items-center justify-center cursor-pointer"
      >
        <div>
          <svg />
        </div>
        <div
          class="ml-[8px]"
        >
          Gaming
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
